connection.go

```diff
@@ -244,6 +249,17 @@ func (sc *snowflakeConn) PrepareContext(ctx context.Context, query string) (driv
 }
 
 func (sc *snowflakeConn) ExecContext(ctx context.Context, query string, args []driver.NamedValue) (driver.Result, error) {
+	qid, _ := getResumeQueryID(ctx)
+	if qid == "" {
+		return sc.execContextInternal(ctx, query, args)
+	}
+	actualQuery := "select * from table(result_scan(" + qid + "))"
+	newCtx := context.WithValue(ctx, FetchResultByID, "")
+	return sc.execContextInternal(newCtx, actualQuery, args)
```

Comment:
the API behavior is going to be a little weird if:
- client passes in a SQL query string
- client also passes in non-empty `args`
- client defines a resume query ID in context

We take the resume query id and feed client's passed in `args` into our substituted SQL `select * from table(...`, this feels like it could be trouble. The way that we do the `qid` string interpolation feels like it could also open us up for potential security hole because it doesn't go through any SQL sanitization.

`ExecContext` is meant for executing SQL and not expecting any result rows, in this case, do we need to expose the QID resuming capability if the SQL is already meant to NOT produce any results?

@sfc-gh-anezvigin if we don't need QID resuming capability from `ExecContext`, I'd advocate keeping our change surface area smaller, and NOT exposing the ability for a client to feed a resume QID into `ExecContext`.
Especially since here we're also resuming by issuing yet another job to do `select * from table(result_scan(...))` which effectively generates a brand new separate job, just to resume a prior SQL that has no result set.